torch_frame/utils/infer_stype.py

```diff
@@ -138,18 +139,25 @@ def infer_series_stype(ser: Series) -> stype | None:
 
             # Try different possible seps and mick the largest min_count.
             min_count_list = []
-            for sep in POSSIBLE_SEPS:
-                try:
-                    min_count_list.append(
-                        _min_count(
-                            ser.apply(lambda row: MultiCategoricalTensorMapper.
-                                      split_by_sep(row, sep)).explode()))
-                except Exception as e:
-                    logging.warn(
-                        "Mapping series into multicategorical stype "
-                        f"with separator {sep} raised an exception {e}")
-                    continue
-            if max(min_count_list) > cat_min_count_thresh:
+            if isinstance(ser.iloc[0], list) or isinstance(
+                    ser.iloc[0], np.ndarray):
+                min_count_list.append(_min_count(ser.explode()))
+            else:
+                for sep in POSSIBLE_SEPS:
+                    try:
+                        min_count_list.append(
+                            _min_count(
+                                ser.apply(
+                                    lambda row: MultiCategoricalTensorMapper.
+                                    split_by_sep(row, sep)).explode()))
+                    except Exception as e:
+                        logging.warn(
+                            "Mapping series into multicategorical stype "
+                            f"with separator {sep} raised an exception {e}")
+                        continue
```

Comment:
nit: I would put `min_count_list` inside else:
then do `max(min_count_list)` in else statement. in the if-statement, I would just put the value of `_min_count(ser.explode())`.

Then judge if the value is larger than `cat_min_count_thresh`

Reply:
got it. Fixed